pkg/phlaredb/delta.go

```diff
@@ -42,7 +42,7 @@ func (d *deltaProfiles) computeDelta(ps *schemav1.Profile, lbs phlaremodel.Label
 	if !ok {
 		// if we don't have the last profile, we can't compute the delta.
 		// so we remove the delta from the list of labels and profiles.
-		d.highestSamples[ps.SeriesFingerprint] = ps.Samples
+		d.highestSamples[ps.SeriesFingerprint] = copySampleSlice(ps.Samples)
```

Comment:
Copying in ensure we don't retain a many sample but just what we need.

This also not retain labels because we don't need them for computing delta.